File: Plot-the-Pattern/C_Protection_Against_Evil_Forces_and_Energies/Protection_Against_Evil_Forces_and_Energies.cpp
#include <bits/stdc++.h>
using namespace std;
string generate(int cnt, int n)
{
    string ans;
    if (2 * cnt >= n)
    {
        for (int i = 0; i < n; i++)
        {
            ans += '*';
            ans += ' ';
        }
        return ans;
    }
    else
    {
        for (int i = 0; i < cnt; i++)
        {
            ans += '*';
            ans += ' ';
        }
        for (int i = 0; i < n - 2 * cnt; i++)
        {
            ans += ' ';
            ans += ' ';
        }
        for (int i = 0; i < cnt; i++)
        {
            ans += '*';
            ans += ' ';
        }
    }
    return ans;
}
int main()
{
    int n;
    cin >> n;
    vector<string> v1;
    int startvalue = (n / 2) + 1;
    for (int i = startvalue; i >= 1; i--)
    {
        cout << generate(i, n) << "\n";
    }
    for (int i = 2; i <= startvalue; i++)
    {
        cout << generate(i, n) << "\n";
    }
}

File: Plot-the-Pattern/C_Protection_Against_Evil_Forces_and_Energies/Protection_Against_Evil_Forces_and_Energies.py
def generate(cnt, n):
    result = ""
    
    if 2 * cnt >= n:
        result = "* " * n
    else:
        result += "* " * cnt
        result += "  " * (n - 2 * cnt)
        result += "* " * cnt
    
    return result

n = int(input())
pattern = []

start_value = (n // 2) + 1
for i in range(start_value, 0, -1):
    pattern.append(generate(i, n))

for i in range(2, start_value + 1):
    pattern.append(generate(i, n))

for line in pattern:
    print(line)
